### PR TITLE
Change retry timeout to get zzk connection

### PR DESCRIPTION
**Pull request by glujan**

### CODE REVIEW

File: zzk/connection.go [hunk 1]
```diff
@@ -181,6 +207,9 @@ func (zconn *zconn) monitor(path string) {
 			return
 		}
 
+		delayC = getExponentialDelay()
+		defer close(delayC)
```
Comment (glujan) [inline zzk/connection.go]: Will this channel be properly closed in each loop run?

**Comment by glujan** [comment]
@smousa Please have a look at this

**Comment by smousa** [comment]
have you looked at this library: https://github.com/jpillora/backoff

5 minutes may be too long for a max backoff.  10-15 seconds seems more reasonable.

**Comment by glujan** [comment]
Previous solution was over-engineered so here is simplified one

File: zzk/connection.go [hunk 2]
```diff
@@ -38,6 +40,34 @@ var (
 // GetConnection describes a generic function for acquiring a connection object
 type GetConnection func(string) (client.Connection, error)
 
+type delay struct {
+	backoff time.Duration
+}
+
+func (d *delay) GetDelay() time.Duration {
+	defer func() {
```
Comment (smousa) [inline zzk/connection.go]: no need to put this in a defer

File: zzk/connection.go [hunk 3]
```diff
@@ -25,6 +26,7 @@ import (
 
 const (
 	DefaultConnectionTimeout = time.Minute
```
Comment (smousa) [inline zzk/connection.go]: Is this still used anywhere?

File: zzk/connection.go [hunk 4]
```diff
@@ -38,6 +40,34 @@ var (
...
+func (d *delay) GetDelay() time.Duration {
+	defer func() {
+		factor := 1.25
+		jitter := 0.1
+
+		d.backoff = time.Duration(float64(d.backoff) * factor)
+		if d.backoff > MaxDelay {
+			d.backoff = MaxDelay
+		}
+		d.backoff += time.Duration(rand.NormFloat64() * jitter * float64(time.Second))
```
Comment (smousa) [inline zzk/connection.go]: This value will stop when it is greater than MaxBackoff
We want it to always be less than or equal to MaxDelay

File: zzk/connection.go (hunk 2 above)
Comment (glujan) [inline zzk/connection.go]: While creating _delay_ object user needs to give _baseDelay_ as argument. I want to return the same value for the first time when _GetDelay_ is being called, thus this `defer` is here

File: zzk/connection.go (hunk 4 above)
Comment (glujan) [inline zzk/connection.go]: Right, I will fix this

File: zzk/connection.go (hunk 2 above)
Comment (smousa) [inline zzk/connection.go]: Please add a comment explaining that.

Comment (glujan) [inline zzk/connection.go]: Sure

**Comment by glujan** [comment]
Retest this please

**Pull request merged**